mutiny-core/src/nostr/mod.rs

```diff
@@ -1285,6 +1262,31 @@ impl<S: MutinyStorage> NostrManager<S> {
     }
 }
 
+pub fn derive_nostr_key<C: Signing>(
+    context: &Secp256k1<C>,
+    xprivkey: ExtendedPrivKey,
+    account: u32,
+    chain: Option<u32>,
+    index: Option<u32>,
+) -> Result<Keys, MutinyError> {
+    let chain = match chain {
+        Some(chain) => ChildNumber::from_hardened_idx(chain)?,
+        None => ChildNumber::from_normal_idx(0)?,
+    };
+
+    let index = match index {
+        Some(index) => ChildNumber::from_hardened_idx(index)?,
+        None => ChildNumber::from_normal_idx(0)?,
+    };
+
+    let path = DerivationPath::from_str(&format!("m/44'/1237'/{account}'/{chain}/{index}"))?;
+    let key = xprivkey.derive_priv(context, &path)?;
+
+    // just converting to nostr secret key, unwrap is safe
+    let secret_key = SecretKey::from_slice(&key.private_key.secret_bytes()).unwrap();
```

Comment:
while you're here, I think we can remove this now

Reply:
Remove what? We use this in multiple places

Reply:
we dont need this line to convert types, this was from us and rust-nostr using different versions of rust-bitcoin

Reply:
okay changed